hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/files/StreamFileProducerBase.java

```diff
@@ -0,0 +1,355 @@
+                    serializedSingleTransactionRecords) {
+                final int numOfSidecarItems =
+                        serializedSingleTransactionRecord.sideCarItemsBytes().size();
+                for (int i = 0; i < numOfSidecarItems; i++) {
+                    // get the sidecar record
+                    final Bytes sidecarRecordBytes = serializedSingleTransactionRecord
+                            .sideCarItemsBytes()
+                            .get(i);
+                    // get the serialized bytes
+                    final TransactionSidecarRecord sideCarItem =
+                            serializedSingleTransactionRecord.sideCarItems().get(i);
+
+                    // get the kind of the sidecar record
+                    final var kind = sideCarItem.sidecarRecords().kind();
+                    // check if we have a latestSidecarWriter, if not create one
+                    if (latestSidecarWriter == null) {
+                        // create a new writer
+                        latestSidecarWriter = new SidecarFileWriter(
+                                getSidecarFilePath(
+                                        currentRecordFileFirstTransactionConsensusTimestamp,
+                                        sidecarFileWriters.size() + 1),
+                                compressFilesOnCreation,
+                                maxSideCarSizeInBytes);
+                        // add writer to list
+                        sidecarFileWriters.add(latestSidecarWriter);
+                    }
+                    // try writing to the latest sidecar file writer
+                    final boolean wasWritten =
+                            latestSidecarWriter.writeTransactionSidecarRecord(kind, sidecarRecordBytes);
+                    // if it was not written then the file is full, so create a new one and write to it
+                    if (!wasWritten) {
+                        // close existing sidecar file writer
+                        latestSidecarWriter.close();
+                        // create a new writer
+                        latestSidecarWriter = new SidecarFileWriter(
+                                getSidecarFilePath(
+                                        currentRecordFileFirstTransactionConsensusTimestamp,
+                                        sidecarFileWriters.size() + 1),
+                                compressFilesOnCreation,
+                                maxSideCarSizeInBytes);
+                        // add writer to list
+                        sidecarFileWriters.add(latestSidecarWriter);
+                        // now write item to the new writer
+                        latestSidecarWriter.writeTransactionSidecarRecord(kind, sidecarRecordBytes);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
What is the right thing for us to do here? We should probably log, for sure. But do we retry? What will happen to the node if an exception happens here? What will happen to the HandleWorkflow? It is hard to follow in the `StreamFileProducerConcurrent`, but I guess it could cause a lot of mayhem. We're constructing long chains of concurrent futures, and the failure may cascade quite far. I'm not sure if it ultimately escapes to cause exceptions in other parts of the code or ends up swallowed in the completable future chain.